shared/modules/error.ts

```diff
@@ -11,13 +11,19 @@ import log from 'loglevel';
  */
 export function isErrorWithMessage(
   error: unknown,
-): error is { message: string } {
-  return typeof error === 'object' && error !== null && 'message' in error;
+): error is { message: string } & { data: { cause: { message: string } } } {
```

Comment:
Hmm, wouldn't this be the correct type? It's one or the other, not necessarily both
```suggestion
): error is { message: string } | { data: { cause: { message: string } } } {
```